### PR TITLE
obs-text: Add 'refresh time' setting when reading from a file

### PR DESCRIPTION
**Pull request by 2003cah**
### Description

This adds a new "Refresh Time" setting in GDI+ text sources when reading from a file. This determines how quickly a change to a text file will reflect on the source, with a minimum of 0 seconds, a maximum of 10 seconds, and a default of 1 second (what it currently is in current builds of OBS).
from this
to here

### Motivation and Context

This was initially sparked by an issue I was having. I'm planning to run a music ticker on my stream, but the 1 second of artificial delay for it to update was a tiny bit experience-ruining when songs changed (as the previous song's title would still be there as the new song was starting). My desire to change this eventually lead me to an [OBS forum thread](https://obsproject.com/forum/threads/read-from-text-file-refresh-rate.73589/) where people were wanting the same capability of changing the delay. 

### How Has This Been Tested?

Using Windows 10 (build 19042) and frantically spamming file writes (through the software I use for the aforementioned ticker), I have found very few conflicts or issues with this addition.

The only exception is that doing file writes too quickly (many times a second) can sometimes have OBS stop displaying the source properly, being "stuck" on a particular set of text until the next file write, but I can't see this being an issue in traditional use.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

This will be my first pull request ever, so please yell at me if I blundered something. I am also open to any suggestions.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
As you noted this has the same bug as the rest of the attempts to increase the refresh rate and which the "artificial" delay was introduced to fix. There is still no desire to accept half hearted attempts at increasing refresh rate (at least by me).

You can always compile this in yourself if you need it or you can write a simple script to achieve the same effect without the aforementioned bug like https://gist.github.com/kkartaltepe/861b02882056b464bfc3e0b329f2f174

**Comment by WizardCM** [comment]
Yeah the correct, proper fix (and the only one I believe would be merged) would be one that actually listens to file changes rather than working on a timer.

**Comment by dodgepong** [comment]
Please see the discussion last time someone tried to change the file read time: https://github.com/obsproject/obs-studio/pull/3010 (contains links to the last several PRs that have tried to implement this feature)

tl;dr The correct way to fix this is to change the file updater to be event-based rather than polling, as @WizardCM said.
